integrationtests/src/test/java/io/kroxylicious/proxy/ClusterIT.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.DescribeClusterResult;
+import org.apache.kafka.clients.admin.KafkaAdminClient;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.kroxylicious.proxy.testcluster.Cluster;
+import io.kroxylicious.proxy.testcluster.ClusterConfig;
+import io.kroxylicious.proxy.testcluster.ClusterFactory;
+import io.kroxylicious.proxy.testcluster.ContainerBasedKafkaCluster;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+/**
+ * Test case that simply exercises the ability to control the kafka cluster from the test.
+ *
+ */
+public class ClusterIT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClusterIT.class);
+
+    @Test
+    public void kafkaClusterKraftMode() throws Exception {
+        try (var cluster = ClusterFactory.create(ClusterConfig.builder().kraftMode(true).build())) {
+            cluster.start();
+            verify(1, cluster);
+        }
+    }
+
+    @Test
+    public void kafkaClusterZookeeperMode() throws Exception {
+        try (var cluster = ClusterFactory.create(ClusterConfig.builder().kraftMode(false).build())) {
+            cluster.start();
+            verify(1, cluster);
+        }
+    }
+
+    @Test
+    public void kafkaMultiNodeClusterKraftMode() throws Exception {
+        int brokersNum = 2;
+        try (var cluster = ClusterFactory.create(ClusterConfig.builder().brokersNum(brokersNum).kraftMode(true).build())) {
+            assumeTrue(cluster instanceof ContainerBasedKafkaCluster, "FIXME: kraft timing out on shutdown in multinode case");
```

Comment:
It's embarrassing, this is a bug in Kafka. We don't expect combined mode (i.e. process.role = broker, controller) used in multi nodes case. I think that's a bad assumption. I'll create a bug in upstream Kafka.

Reply:
[KAFKA-14287](https://issues.apache.org/jira/browse/KAFKA-14287) is created for this issue.

Reply:
For now, I think we can either skip this test case, or separate controller and broker in different nodes.

Reply:
Thanks @showuon